File: solutions/go/bob/1/bob.go
package bob

import "strings"

func isQuestion(sentence string) bool {
	return strings.HasSuffix(sentence, "?")
}
func isYelling(sentence string) bool {
	return strings.ToUpper(sentence) == sentence && strings.ToLower(sentence) != sentence
}
func isQuestionYelling(sentence string) bool {
	return isQuestion(sentence) && isYelling(sentence)
}
func isEmpty(sentence string) bool {
	return sentence == ""
}
func Hey(remark string) string {
	remark = strings.TrimSpace(remark)
	switch {
	case isQuestionYelling(remark):
		return "Calm down, I know what I'm doing!"
	case isQuestion(remark):
		return "Sure."
	case isYelling(remark):
		return "Whoa, chill out!"
	case isEmpty(remark):
		return "Fine. Be that way!"
	default:
		return "Whatever."
	}
}

File: solutions/cpp/bob/1/bob.cpp
#include "bob.h"

namespace bob {
    std::string hey(const std::string& s)
    {
        bool question = false;
        bool upperCase = false;
        bool lowerCase = false;
        bool nothing = true;
        
        for (char c : s) {
            if (islower(c))
                lowerCase = true;
            if (isupper(c))
                upperCase = true;
            if (!isspace(c)) {
                question = false;
                nothing = false;
            }
            if (c == '?')
            question = true;
        }
        
        bool shouting = upperCase && !lowerCase;
        if (shouting && question)
            return "Calm down, I know what I'm doing!";
        if (shouting)
            return "Whoa, chill out!";
        if (question)
            return "Sure.";
        if (nothing)
            return "Fine. Be that way!";
        return "Whatever.";
    }
}  // namespace bob
